src/uwtools/cli.py

```diff
@@ -305,6 +309,132 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
     )
 
 
+# Mode Rocoto
+
+
+def _add_subparser_rocoto(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: rocoto
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.rocoto, "Realize and validate Rocoto XML Documents")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.submode)
+    return {
+        STR.realize: _add_subparser_rocoto_realize(subparsers),
+        STR.validate: _add_subparser_rocoto_validate(subparsers),
+    }
+
+
+def _add_subparser_rocoto_realize(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto realize
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.realize, "Realize a Rocoto XML workflow document")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_input_file(required)
+    _add_arg_output_file(required)
+    optional = _basic_setup(parser)
+    _add_arg_input_format(optional, choices=[FORMAT.yaml])
+    _add_arg_output_format(optional, choices=[FORMAT.rocoto])
```

Comment:
These are not actually optional. Input is always YAML, output is always XML.

Reply:
Hmm, there are "optional-but-one-choice" settings in other CLI functions, I should probably make them fully fixed as well

Reply:
Hm, on re-look, this looks like it is just setting up `_check_file_vs_format` to ensure the input file is the correct format. I could